### PR TITLE
PIM-5483: Improve loading of product datagrid

### PR DESCRIPTION
**Pull request by skeleton**
| Q                 | A
| ----------------- | ---
| Specs             | Y
| Behats            | Y
| Blue CI           | ?
| Changelog updated | Y
| Review and 2 GTM  |

This PR contains cherry-pick from 1.4

### CODE REVIEW

**Comment by jjanvier** [comment]
gtm

File: features/product/associate_product.feature [hunk 1]
```diff
@@ -70,3 +70,97 @@ Feature: Associate a product
     Then the row "shoelaces" should be checked
     And the row "black-boots" should be checked
     And I should be able to sort the rows by IS ASSOCIATED
+    And I should be able to sort the rows by SKU
+
+  @skip @jira https://akeneo.atlassian.net/browse/PIM-4670
```
Comment (juliensnz) [inline features/product/associate_product.feature]: you add a behat and skip it right away ?

**Comment by grena** [comment]
gtm

**Comment by pierallard** [comment]
gtm

File: features/product/associate_product.feature (hunk 1 above)
Comment (skeleton) [inline features/product/associate_product.feature]: indeed... it's a cherry-pick but I can remove it

File: src/Pim/Bundle/DataGridBundle/Resources/views/macros.html.twig [hunk 2]
```diff
@@ -42,6 +42,10 @@
                         urlParams['{{ name }}[_parameters][view][columns]'] = state.columns;
                     }
 
+                    {% if name == 'product-grid' %}
+                    var categoryFilter = new CategoryFilter(urlParams, '{{ name }}');
```
Comment (juliensnz) [inline src/Pim/Bundle/DataGridBundle/Resources/views/macros.html.twig]: could you indent this line ?

**Comment by juliensnz** [comment]
GTM :) I throw up a couple of time but it was not your fault so I swallowed it back and dealt with it

**Pull request merged**